### PR TITLE
Cleaned TS yaml rules

### PR DESCRIPTION
**Pull request by reaperrr**
Streamlines actor defaults.

### CODE REVIEW

**Comment by GraionDilach** [comment]
You coulda enable the EMP sparkle anim as part of this iMO - now that we have WithDecoration with HCenter-VCenter positioning.

File: mods/ts/rules/defaults.yaml [hunk 1]
```diff
@@ -271,50 +258,57 @@
...
 	MustBeDestroyed:
+	TakeCover:
+		DamageModifiers:
+			Prone50Percent: 50
+			Prone60Percent: 60
+			Prone70Percent: 70
+			Prone100Percent: 100
+			Prone350Percent: 350
+		DamageTriggers: TriggerProne
+	TakeCover:
```
Comment (abcdefg30) [inline mods/ts/rules/defaults.yaml]: Another `TakeCover:`? (see line 272)

Comment (reaperrr) [inline mods/ts/rules/defaults.yaml]: Silly copy-paste error that I somehow overlooked.

**Comment by abcdefg30** [comment]
Needs a rebase.

**Comment by reaperrr** [comment]
Rebased.

**Comment by reaperrr** [comment]
I want to split off the third commit into a separate PR that will deal with civ building sequence offsets as well.

**Comment by pchote** [comment]
The mammoth mk II and carryall selection boxes are weird, but everything else looks fine.  :+1: after fixing those.

**Comment by reaperrr** [comment]
Updated.

fixed-selectionboxes

**Comment by obrakmann** [comment]
I suspect this means something's wrong with the viceroids:

```
Exception of type `System.InvalidOperationException`: Actor has multiple traits of type `OpenRA.Mods.Common.Traits.RenderSprites`
  at OpenRA.TraitDictionary+TraitContainer`1[OpenRA.Mods.Common.Traits.RenderSprites].GetOrDefault (UInt32 actor) [0x000b4] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:161 
  at OpenRA.TraitDictionary+TraitContainer`1[OpenRA.Mods.Common.Traits.RenderSprites].Get (UInt32 actor) [0x00003] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:146 
  at OpenRA.TraitDictionary.Get[RenderSprites] (OpenRA.Actor actor) [0x00013] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:96 
  at OpenRA.Actor.Trait[RenderSprites] () [0x0000d] in /home/oliver/devel/openra/git/OpenRA.Game/Actor.cs:212 
  at OpenRA.Mods.Common.Traits.WithSpriteBody..ctor (OpenRA.ActorInitializer init, OpenRA.Mods.Common.Traits.WithSpriteBodyInfo info, System.Func`1 baseFacing) [0x00027] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:55 
  at OpenRA.Mods.Common.Traits.WithSpriteBody..ctor (OpenRA.ActorInitializer init, OpenRA.Mods.Common.Traits.WithSpriteBodyInfo info) [0x00000] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:50 
  at OpenRA.Mods.Common.Traits.WithSpriteBodyInfo.Create (OpenRA.ActorInitializer init) [0x00003] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:29 
  at OpenRA.Actor..ctor (OpenRA.World world, System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x000c8] in /home/oliver/devel/openra/git/OpenRA.Game/Actor.cs:95 
  at OpenRA.World.CreateActor (Boolean addToWorld, System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x00004] in /home/oliver/devel/openra/git/OpenRA.Game/World.cs:241 
  at OpenRA.World.CreateActor (System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x00005] in /home/oliver/devel/openra/git/OpenRA.Game/World.cs:236 
  at OpenRA.Mods.Cnc.Traits.SpawnViceroid+<Killed>c__AnonStorey0.<>m__0 (OpenRA.World w) [0x0006e] in /home/oliver/devel/openra/git/OpenRA.Mods.Cnc/Traits/SpawnViceroid.cs:51
```

File: mods/ts/rules/defaults.yaml [hunk 2]
```diff
@@ -376,84 +373,40 @@
...
+	WithVoxelBody:
+
+^VoxelTank:
+	Inherits: ^Tank
+	RenderVoxels:
+	WithVoxelBody:
+
+^CivilianVoxelVehicle:
+	Inherits: ^VoxelVehicle
+	-RenderSprites: # They don't have icons so this is needed to satisfy --check-yaml
```
Comment (penev92) [inline mods/ts/rules/defaults.yaml]: I vote for avoiding comments in YAML wherever possible, because the utility doesn't like them and keeps removing them. Also I don't think this is exactly needed here.

File: mods/ts/rules/defaults.yaml [hunk 3]
```diff
@@ -1,3 +1,14 @@
+^ExistsInWorld:
+	AppearsOnRadar:
+	UpdatesPlayerStatistics:
+	CombatDebugOverlay:
+	DrawLineToTarget:
```
Comment (penev92) [inline mods/ts/rules/defaults.yaml]: This looks like an overkill, considering not even half of the actors that inherit this will need target lines.

**Comment by reaperrr** [comment]
Updated.

- fixed visceroid crash
- removed comment from civ voxel vehicle default
- moved DrawLineToTarget to moving actors so buildings won't inherit it

- slightly reorganised aircraft defaults

**Comment by reaperrr** [comment]
I'll squash the last commit when this has its 2nd +1.

File: mods/ts/rules/defaults.yaml [hunk 4]
```diff
@@ -41,18 +51,33 @@
...
 		Adjacent: 4
 	FrozenUnderFog:
-	GivesBuildableArea:
-	Capturable:
 	SoundOnDamageTransition:
 		DamagedSounds: expnew01.aud
 		DestroyedSounds: crmble2.aud
 	RenderBuilding:
 	WithBuildingExplosion:
+	RepairableBuilding:
```
Comment (GraionDilach) [inline mods/ts/rules/defaults.yaml]: This lets civilian buildings be repaired via the mouse. I thought they were only repairable via engineers in TS as well.

Comment (reaperrr) [inline mods/ts/rules/defaults.yaml]: You're right, I overlooked that. Will fix it.

File: mods/ts/rules/defaults.yaml [hunk 5]
```diff
@@ -41,19 +51,34 @@
...
-	GivesBuildableArea:
-	Capturable:
 	SoundOnDamageTransition:
 		DamagedSounds: expnew01.aud
 		DestroyedSounds: crmble2.aud
 	RenderBuilding:
 	WithBuildingExplosion:
+	EngineerRepairable:
+	ShakeOnDeath:
+	ProximityCaptor:
```
Comment (GraionDilach) [inline mods/ts/rules/defaults.yaml]: Why is this needed?

Comment (penev92) [inline mods/ts/rules/defaults.yaml]: That one isn't needed in any of the official mods. Some Capture the Flag maps use it, but they can define it for themselves.

**Comment by reaperrr** [comment]
Removed ProximityCaptor and squashed the last 3 smaller commits.

**Comment by Mailaender** [comment]
> Removes bogus TEMPERATE exclusivity from CA00** civ buildings.

I still don't see them on SNOW.

**Comment by reaperrr** [comment]
> I still don't see them on SNOW.

Right, I decided to concentrate on the actor defaults for this PR and moved that fix to an upcoming follow-up PR quite a while ago (that follow-up will fix civilian structure sequence offsets as well), forgot to update the PR description.

**Comment by Mailaender** [comment]
Can't spot any regressions in-game. :white_check_mark:

**Comment by GraionDilach** [comment]
Even if I come late to the party but I really welcome this cleanup. It looks so much better now. Great job. :)
